include/pybind11/pybind11.h

```diff
@@ -439,9 +439,11 @@ class cpp_function : public function {
 
         /* Install docstring */
         auto *func = (PyCFunctionObject *) m_ptr;
-        if (func->m_ml->ml_doc)
-            std::free(const_cast<char *>(func->m_ml->ml_doc));
-        func->m_ml->ml_doc = strdup(signatures.c_str());
+        std::free(const_cast<char *>(func->m_ml->ml_doc));
+        func->m_ml->ml_doc = nullptr;
```

Comment:
Couldn’t this be in an else branch below the next line? Or maybe oven better, make it a ternary operator?

Reply:
Yeah, I just made it a ternary operator... and I almost forget that ternary operator is lazily evaluated like the if statement in C++. Haven't been writing C++ for some time...